artificial-scrum-master/apps/artificial.scrum.master/Artificial.Scrum.Master.ScrumProjectIntegration/Infrastructure/ScrumServiceHttpClient/ProjectHttpClientWrapper.cs

```diff
@@ -0,0 +1,120 @@
+using System.Net;
+using System.Text.Json;
+using Artificial.Scrum.Master.ScrumProjectIntegration.Exceptions;
+using Artificial.Scrum.Master.ScrumProjectIntegration.Infrastructure.ApiTokens;
+using Artificial.Scrum.Master.ScrumProjectIntegration.Infrastructure.Models;
+using Artificial.Scrum.Master.ScrumProjectIntegration.Settings;
+using Artificial.Scrum.Master.ScrumProjectIntegration.Utilities;
+
+namespace Artificial.Scrum.Master.ScrumProjectIntegration.Infrastructure.ScrumServiceHttpClient
+{
+    public class ProjectHttpClientWrapper : IProjectHttpClientWrapper
+    {
+        private readonly ScrumManagementServiceSettings _settings;
+
+        private readonly IHttpClientFactory _httpClientFactory;
+        private readonly IJwtDecoder _jwtDecoder;
+        private readonly IUserTokensRepository _userTokensRepository;
+
+        public ProjectHttpClientWrapper(ScrumManagementServiceSettings settings, IHttpClientFactory httpClientFactory,
+            IUserTokensRepository userTokensRepository, IJwtDecoder jwtDecoder)
+        {
+            _httpClientFactory = httpClientFactory;
+            _userTokensRepository = userTokensRepository;
+            _jwtDecoder = jwtDecoder;
+            _settings = settings;
+        }
+
+        public async Task<WrapperHttpResponse> GetHttpRequest(string userId, string url)
+        {
+            var tokens = await _userTokensRepository.GetUserAccessTokens(userId);
+            if (tokens is null)
+            {
+                throw new ProjectResourceNotFoundException($"Credentials of user:{userId} not found");
+            }
+
+            var isValid = ValidateAccessToken(tokens.Value.AccessToken);
+            if (!isValid)
+            {
+                tokens = await RefreshUserTokens(userId, tokens.Value);
+            }
+
+            var httpClient = _httpClientFactory.CreateClient();
+            httpClient.DefaultRequestHeaders.Add("Authorization", $"Bearer {tokens.Value.AccessToken}");
+
+            var httpResponse = await httpClient.GetAsync(url);
+            var result = await httpResponse.Content.ReadAsStringAsync();
```

Comment:
bardzo mi sie to nie podoba. Czemu to zwraca stringa? Nie możesz zrobić tych metod generycznych i zrobić GetFromJsonAsync<T>(url)? Dostajesz od razu obiekt jsonowy. i te 2 linijki masz w jednej + nie zwracasz syfo stringa